mmgen/models/architectures/vqgan/modules.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn.bricks import (Linear, build_activation_layer, build_conv_layer,
+                             build_norm_layer)
+from mmgen.registry import MODULES
+
+
+def nonlinearity(x):
+    """Sigmoid function.
+
+    Args:
+        x (torch.Tensor): Input feature map.
+
+    Returns:
+        torch.Tensor: Output feature map.
+    """
+    activate = build_activation_layer(dict(type='Sigmoid'))
+    return x * activate(x)
+
+
+def Normalize(in_channels):
+    """Normalize function.
+
+    Args:
+        in_channels (int): The channel number of the input feature map.
+
+    Returns:
+        norm(torch.FloatTensor): Output feature map.
+    """
+    norm_cfg = dict(type='GN', num_groups=32, eps=1e-6, affine=True)
+    _, norm = build_norm_layer(norm_cfg, in_channels)
+    return norm
+
+
+@MODULES.register_module()
+class DiffusionDownsample(nn.Module):
```

Comment:
I'm wondering whether we call this module ``DiffusionDownsample``.😂

Reply:
This module is a single stride2 conv or avg_pool. We may not add an extra class here.